README.md

```diff
@@ -1,44 +1,99 @@
 # puppetwash
 
-A [Wash](https://puppetlabs.github.io/wash/) plugin for Puppet Enterprise. It will
-iterate all nodes from PuppetDB and expose information about them, such as their
-facts, last run report, and the current catalog.
+A [Wash](https://puppetlabs.github.io/wash/) plugin for managing your Puppet instances
+(both PE and open source). This plugin will iterate all nodes from PuppetDB and expose
+information about them such as their facts, last run report, and the current catalog.
 
-## Installation and configuration
+```
+wash . ❯ stree puppet
+puppet
+└── [instance]
+    └── nodes
+        └── [node]
+            ├── catalog
+            ├── facts
+            │   └── [fact]
+            └── reports
+                └── [report]
+```
 
-1. clone this repo and cd to it
-2. run `bundle install`
-3. Add to `~/.puppetlabs/wash/wash.yaml`:
+Here's an example that shows off the `puppetwash` plugin's features.
 
-    ```yaml
-    external-plugins:
-        - script: '/path/to/puppetwash/puppet'
-    ```
+```
```

Comment:
Might be valuable to break this up a bit and do the description in a separate paragraph. People tend to gloss over large blocks of text.

Reply:
Doesn't look too bad when formatted -- https://github.com/ekinanp/puppetwash/tree/fixes. If I were to do this, what format do you have in mind for the README?

Reply:
Just something like...

The Puppet plugin lists all nodes that have reported in to PuppetDB
```
wash . ❯ ls puppet/instance_one/nodes
base-graham.delivery.puppetlabs.net/
choosy-cat.delivery.puppetlabs.net/
fugual-apology.delivery.puppetlabs.net/
lunar-scoundrel.delivery.puppetlabs.net/
sturdy-romp.delivery.puppetlabs.net/
```
You can view the last catalog they applied
```
wash . ❯ cat puppet/instance_one/nodes/base-graham.delivery.puppetlabs.net/catalog.json
{
  "catalog_uuid": "663da32b-b146-4d7a-8367-52849cb4ed5f",
  "producer": "sturdy-romp.delivery.puppetlabs.net",
  "hash": "ae8bef9977cd10c85a949cb8af97f3d27739bdc9",
...
```
Or find all reports that were created within the last 10 hours
```
wash . ❯ find puppet -k '*report' -mtime -10h
puppet/instance_one/nodes/base-graham.delivery.puppetlabs.net/reports/2020-02-07T18:26:31.789Z
puppet/instance_one/nodes/choosy-cat.delivery.puppetlabs.net/reports/2020-02-07T18:26:30.917Z
puppet/instance_one/nodes/choosy-cat.delivery.puppetlabs.net/reports/2020-02-07T18:30:05.915Z
...
wash . ❯ cat puppet/instance_one/nodes/base-graham.delivery.puppetlabs.net/reports/2020-02-07T18\:26\:31.789Z
[
  {
    "catalog_uuid": "663da32b-b146-4d7a-8367-52849cb4ed5f",
    "receive_time": "2020-02-07T18:26:32.375Z",
    "producer": "sturdy-romp.delivery.puppetlabs.net",
    "hash": "9a0ea00afed5b00e0d6fe8130534c6dda08beccc",
    "transaction_uuid": "d103e05c-1bd7-41eb-a333-cea4fb26fd57",
...
```
It's also easy to search for specific facts since they're listed as individual files
```
wash . ❯ ls puppet/instance_one/nodes/base-graham.delivery.puppetlabs.net/facts
aio_agent_build
aio_agent_version
architecture
augeas
augeasversion
bios_release_date
...
wash . ❯ cat puppet/instance_one/nodes/base-graham.delivery.puppetlabs.net/facts/aio_agent_build
6.12.0.177.g77ef0268
```

Reply:
Ahh that's what you were referring to. Ok, I can split that up.